src/main/java/cz/cvut/kbss/analysis/controller/ComponentController.java

```diff
@@ -59,29 +80,56 @@ public void delete(@PathVariable(name = "componentFragment") String componentFra
         repositoryService.remove(componentUri);
     }
 
+    @Operation(summary = "Retrieve functions of a component")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Functions retrieved"),
+            @ApiResponse(responseCode = "404", description = "Component not found")
+    })
     @GetMapping(value = "/{componentFragment}/functions", produces = {JsonLd.MEDIA_TYPE, MediaType.APPLICATION_JSON_VALUE})
     public Set<Function> getFunctions(@PathVariable(name = "componentFragment") String componentFragment) {
         log.info("> getFunctions - {}", componentFragment);
         URI componentUri = identifierService.composeIdentifier(Vocabulary.s_c_component, componentFragment);
         return repositoryService.getFunctions(componentUri);
     }
 
+    @Operation(summary = "Retrieve failure modes of a component")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Failure modes retrieved"),
+            @ApiResponse(responseCode = "404", description = "Component not found")
+    })
     @GetMapping(value = "/{componentFragment}/failureModes", produces = {JsonLd.MEDIA_TYPE, MediaType.APPLICATION_JSON_VALUE})
     public Set<FailureMode> getFailureModes(@PathVariable(name = "componentFragment") String componentFragment) {
         log.info("> getFailureModes - {}", componentFragment);
         URI componentUri = identifierService.composeIdentifier(Vocabulary.s_c_component, componentFragment);
         return repositoryService.getFailureModes(componentUri);
     }
 
+
+    @Operation(summary = "Add new failure mode to a component")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "Failure mode added"),
+            @ApiResponse(responseCode = "400", description = """
+    Invalid Invalid failureMode, e.g.
+    - The uri should be null or unique
+    - Duplicate entity name
+    - Name must not be empty
```

Comment:
```suggestion
    - failure mode uri is not null nor unique.
    - failure mode name already exists.
    - failure mode name is empty.
```

Reply:
I tried to make it more consistent as it seemed to me that some of the bulets have semantics of **errors** (Duplicate entity name) and other ones semantics of **constraints** (The uri should be null or unique)

Reply:
@blcham 
Ok but there are two issues:
- I will have to update the actual messages in the exceptions/validation-errors.
- these messages are generic - I will have to find a way to add the entity type, e.g. `failure mode`) to the message.